LanguageFeatures/Static-access-shorthand/equality_A01_t01.dart

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) 2024, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion If an expression has the form `e1 == e2` or `e1 != e2`, or a
+/// pattern has the form `== e2` or `!= e2`, where the static type of `e1`, or
+/// the matched value type of the pattern, is S1, and e2 is precisely a
+/// `<staticMemberShorthand>` expression, then assign the type S1 as the
+/// shorthand context of the `<staticMemberShorthandHead>` of e2 before
+/// inferring its static type the same way as above.
+///
+/// @description Checks that if an expression has the form `e1 == e2` or
+/// `e1 != e2` and `e2` is a shorthand expression, then it has context type from
```

Comment:
I think it would be helpful to use the terminology from the feature spec:
```suggestion
/// `e1 != e2` and `e2` is a shorthand expression then it has shorthand context from
```
The point is that it isn't actually the context type for `e2`. The context type for `e2` is determined by the statically known declaration of operator `==` and static access shorthands won't change anything about that. This means that equality expressions is _the_ topic where we need to make it explicit that we have a context type, and we have a shorthand context, and they are not the same (even though they are the same everywhere else).